integration-core/src/main/java/org/symphonyoss/integration/core/service/CryptoServiceImpl.java

```diff
@@ -93,26 +75,33 @@ public String encrypt(String plainText, String key) throws CryptoException {
     checkParameters("plainText", plainText);
     checkParameters("key", key);
     try {
-      byte[] saltBytes = generateSalt();
-      SecretKeySpec secret = deriveKey(key, saltBytes);
+      // Generate a random salt
+      byte saltBytes[] = new byte[SALT_SIZE];
+      Bytes.random(saltBytes);
 
-      Cipher cipher = getCipher();
-      initCipher(cipher, Cipher.ENCRYPT_MODE, secret, null);
-      AlgorithmParameters params = cipher.getParameters();
-      byte[] ivBytes = params.getParameterSpec(IvParameterSpec.class).getIV();
-      byte[] encryptedTextBytes = doCipher(cipher, plainText.getBytes(CHARSET));
+      // Derive a Password-based key from the key parameter + salt
+      byte[] pbKey = PBKDF.PBKDF2_SHA256(key.getBytes(CHARSET), saltBytes, NUMBER_OF_ITERATIONS);
 
-      // Prepend Salt and VI
-      byte[] buffer = new byte[saltBytes.length + ivBytes.length + encryptedTextBytes.length];
-      System.arraycopy(saltBytes, 0, buffer, 0, saltBytes.length);
-      System.arraycopy(ivBytes, 0, buffer, saltBytes.length, ivBytes.length);
-      System.arraycopy(encryptedTextBytes, 0, buffer, saltBytes.length + ivBytes.length,
-          encryptedTextBytes.length);
+      // Encrypt the plain text
+      KeyIdentifier keyId = new KeyIdentifier(new byte[STREAM_ID_SIZE], 0L, 0L);
```

Comment:
A comment explaining why you create this object with all zeroes will be very helpful for future generations. :)

Reply:
Done.